### PR TITLE
Code Cleanup: Java 16 instanceof pattern matching

### PR DESCRIPTION
**Pull request by SougandhS**
Performs code cleanup in pattern matching using instanceof operator according to Java 16 guidelines, making the code more concise for classes in jdt.launching package.

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by SougandhS** [comment]
Hi @jukzi , could you please check on this ?

**Comment by jukzi** [comment]
Was this an pure automated cleanup, or manual edit? Or both?

**Comment by SougandhS** [comment]
> Was this an pure automated cleanup, or manual edit? Or both?

Manual edit only.

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java [hunk 1]
```diff
@@ -144,17 +144,17 @@ public class JavaLaunchableTester extends PropertyTester {
 	 */
 	private IType getType(IJavaElement element) {
         IType type = null;
-        if (element instanceof ICompilationUnit) {
-            type= ((ICompilationUnit) element).findPrimaryType();
+		if (element instanceof ICompilationUnit iCompilationUnit) {
```
Comment (jukzi) [inline org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java:147]: on github the intendation looks wrong here

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java [hunk 2]
```diff
@@ -3337,8 +3331,8 @@ private static void updateCompliance(IVMInstall vm) {
 		if (LaunchingPlugin.isVMLogging()) {
 			LaunchingPlugin.log("Compliance needs an update."); //$NON-NLS-1$
 		}
-        if (vm instanceof IVMInstall2) {
-            String javaVersion = ((IVMInstall2)vm).getJavaVersion();
+		if (vm instanceof IVMInstall2 ivmInstall) {
```
Comment (jukzi) [inline org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java:3334]: intendation looks wrong on github

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java (hunk 1 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java:147]: in local its ok
image

**Comment by jukzi** [comment]
beside minor intendation looks ok for me

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java (hunk 2 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java:3334]: in local there's no modification any intendations. 
image

Comment (akurtakov) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java:3334]: Enable "show whitespace characters" and you'll most likely see a mixture of tabs and spaces and different tools display tabs as 2,4,8,... whitespaces.

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java (hunk 1 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/JavaLaunchableTester.java:147]: Will do this.

**Comment by jukzi** [comment]
> Manual edit only.

please make that clear in the commit message "**manual** code cleanup" to not be confused with dogfooding

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java (hunk 2 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java:3334]: > Enable "show whitespace characters" and you'll most likely see a mixture of tabs and spaces and different tools display tabs as 2,4,8,... whitespaces.

Thanks

**Comment by SougandhS** [comment]
> > Manual edit only.
> 
> please make that clear in the commit message "**manual** code cleanup" to not be confused with dogfooding

Done the changes now.

**Comment by SougandhS** [comment]
Hi @jukzi 
I have removed all changes from JavaLaunchableTester.java as it is doing some auto formatting once it is saved, resulting huge collapse in indentations

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java [hunk 3]
```diff
@@ -115,102 +114,85 @@
 import org.xml.sax.SAXException;
 
 /**
- * The central access point for launching support. This class manages
- * the registered VM types contributed through the
- * <code>"org.eclipse.jdt.launching.vmType"</code> extension point.
- * As well, this class provides VM install change notification,
- * and computes class paths and source lookup paths for launch
- * configurations.
+ * The central access point for launching support. This class manages the registered VM types contributed through the
```
Comment (jukzi) [inline org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java:117]: Manu unrelated format changes in JavaRuntime.java

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/sourcelookup/containers/JavaProjectSourceContainer.java [hunk 4]
```diff
@@ -119,7 +119,7 @@ protected ISourceContainer[] createSourceContainers() throws CoreException {
 						IPath path = entry.getPath();
 						IResource resource = root.findMember(path);
 						if (resource instanceof IContainer) {
```
Comment (jukzi) [inline org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/sourcelookup/containers/JavaProjectSourceContainer.java:121]: missing cleanup here

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java (hunk 3 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/JavaRuntime.java]: Same case as of JavaLaunchableTester.java - So removed the changes now.

File: org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/sourcelookup/containers/JavaProjectSourceContainer.java (hunk 4 above)
Comment (SougandhS) [reply org.eclipse.jdt.launching/launching/org/eclipse/jdt/launching/sourcelookup/containers/JavaProjectSourceContainer.java]: Done.

**Comment by SougandhS** [comment]
> LGTM, lets wait some days if someone else wants to review too.

Sure @jukzi , Thanks

**Comment by SougandhS** [comment]
Thank you @akurtakov & @jukzi

**Comment by jukzi** [comment]
> I really find such PRs reducing time to read code kind one of the most important things as with the limited resources it's very hard to dedicate time for issues/improvements making every second counts.

I like such motivating comments. thanks. And thanks for @SougandhS for improving the code even in minor cases.

**Comment by SougandhS** [comment]
> I like such motivating comments. thanks. And thanks for @SougandhS for improving the code even in minor cases.

Thanks again @jukzi
